src/EditorFeatures/Core/ExternalAccess/VSTypeScript/Api/VSTypeScriptFormattingInteractionServiceWrapper.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Editor.ExternalAccess.VSTypeScript.Api
+{
+    internal readonly struct VSTypeScriptFormattingInteractionServiceWrapper
+    {
+        private readonly IFormattingInteractionService _underlyingObject;
+
+        private VSTypeScriptFormattingInteractionServiceWrapper(IFormattingInteractionService underlyingObject)
+        {
+            _underlyingObject = underlyingObject;
+        }
+
+        public static VSTypeScriptFormattingInteractionServiceWrapper Create(Document document)
+            => new(document.Project.LanguageServices.GetRequiredService<IFormattingInteractionService>());
```

Comment:
Should this use the `FormattingInteractionServiceProxy`?

Reply:
No. This intentionally using the new type so that it will light up when typescript moves to it.

Reply:
Actually, I think this is entirely unneeded. This wrapper is for TS to get access to _our_ implementation, so I'll just delete this and move their wrapper to use the proxy.